### PR TITLE
[DEVOPS] Add dockerfiles for backend (go) and database (postgres)

### PR DESCRIPTION
**Pull request by LuiggiPasacheL**
# Database credentials:
User: user
Password: password
Database name: social_app

# Ports:
Postgres: [5432](localhost:5432)
Api: [3333](localhost:3333)
Adminer: [8080](localhost:8080)

---
Aditionally, I add the adminer service (localhost:8080) for database management.
Maybe docker-compose file can be placed on the root of the project (to adding frontend service).
This PR is only for backend and database, still need to create dockerfile for frontend and its service in docker compose.

### CODE REVIEW

File: backend/docker-compose.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+      - "3333:3333"
+    depends_on: 
+      - db
+  db:
+    image: postgres:16.0-alpine3.18
+    restart: always
+    environment:
+      POSTGRES_USER: user
+      POSTGRES_PASSWORD: password
+      POSTGRES_DB: social_app
```
Comment (ekimike) [inline backend/docker-compose.yaml:16]: Just bear in mind that this information is sensitive. You can leave it as is by now.

**Review by ekimike — COMMENTED** [review]
The intention of this comment is to consider that security is a serious topic to consider while working on software development.

File: backend/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+WORKDIR /app
+
+COPY . .
+
+RUN go mod download && go mod verify
+
+RUN go build -o build/main cmd/server.go
+
+CMD ["build/main"]
```
Comment (andresrdlc) [inline backend/Dockerfile:11]: Port 3333 needs to be exposed for deployment from Cloud Run. Additionally, I suggest modifying the RUN and CMD command to follow the nomenclature defined in the continuous integration pipeline.

I propose to make the following change:

```FROM golang:1.21.1-alpine3.18

WORKDIR /app

COPY . .

RUN go mod download

RUN go mod verify

RUN go build -o build/backend-develop cmd/server.go

EXPOSE 3333

CMD ["build/backend-develop"]
```

**Review by andresrdlc — CHANGES_REQUESTED** [review]
Branch nomenclature must be followed. Example:
DEVOPS_feature_test

The PR nomenclature begins with:
[DEVOPS] text

Finally, update the Dockerfile as indicated.

File: backend/docker-compose.yaml (hunk 1 above)
Comment (LuiggiPasacheL) [reply backend/docker-compose.yaml:16]: Yes, I think it should be an option to add environment variables to hide sensitive information, but in order not to add files not related to docker I put the credentials in the dockerfile.

**Pull request closed without merging**